SmartDeviceLink/SDLProtocolHeader.m

```diff
@@ -59,8 +58,12 @@ + (__kindof SDLProtocolHeader *)headerForVersion:(UInt8)version {
             return [[SDLV2ProtocolHeader alloc] initWithVersion:version];
         } break;
         default: {
+#ifdef DEBUG
             NSString *reason = [NSString stringWithFormat:@"The version of header that is being created is unknown: %@", @(version)];
-            @throw [NSException exceptionWithName:NSInvalidArgumentException reason:reason userInfo:@{ @"requestedVersion": @(version) }];
+            @throw [NSException exceptionWithName:NSInvalidArgumentException reason:reason userInfo:@{ @"requestedVersion" : @(version) }];
+#else
+            return [[SDLV2ProtocolHeader alloc] initWithVersion:invalidSDLProtocolVersion];
```

Comment:
Would it be better to return `nil`? I'm not sure this is something we want to do in this PR. It's a change that would require some testing.